### PR TITLE
Issue/211 gsp>=318 raise HTTP error #211

### LINKED ISSUE #211: gsp>=318 raise HTTP error
## Detailed Description
If we enter a gsp greater than (or equal) 318 then we should return a safe http exception `204`

## Context
- good to have safe errors

## Possible Implementation
should add it to
- /v0/solar/GB/gsp/pvlive/{gsp_id}
- /v0/solar/GB/gsp/forecast/{gsp_id}
and if GSP_id >= 318 then return HTTP `204` error

### PR DESCRIPTION
**Pull request by petermnhull**
# Pull Request

## Description

Fixes #211 

- Returns 204 on pvlive and forecast endpoints when gsp provided exceeds 318.
- Corrected typo.
- Removed duplicated test.

## How Has This Been Tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration_

Added 4 unit tests covering the scenarios, and tested by running the app locally with placeholder data.

- [x] Yes

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_
NA.

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/gsp.py [hunk 1]
```diff
@@ -129,6 +134,9 @@ def get_forecasts_for_a_specific_gsp(
     logger.info(f"Get forecasts for gsp id {gsp_id} forecast of forecast with only values.")
     logger.info(f"This is for user {user}")
 
+    if gsp_id > GSP_TOTAL:
+        return Response(None, status.HTTP_204_NO_CONTENT)
```
Comment (peterdudfield) [inline src/gsp.py:138]: nice, thanks

**Review by peterdudfield — APPROVED** [review]

**Comment by peterdudfield** [comment]
Thank you @petermnhull for doing this!

**Comment by peterdudfield** [comment]
@all-contributors please add @petermnhull  for code

**Comment by petermnhull** [comment]
Thanks for approving and merging @peterdudfield ! Happy to be able to contribute.

**Pull request merged**